lib/super_digit.rb

```diff
@@ -1,15 +1,18 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n^2)
+# Space Complexity - O(n^2)
 def super_digit(n)
-   
+  n_array = n.to_s.split('').map {|i| i.to_i}
+  return n if n_array.length == 1
+  return super_digit(n_array.sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n^2)
+# Space Complexity - O(n^2)
 def refined_super_digit(n, k)
```

Comment:
This is the brute force solution.  It works, but you can do better with dynamic programming.